omp/matrix/csr_kernels.cpp

```diff
@@ -742,6 +744,55 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_CALC_NNZ_PER_ROW_IN_SPAN_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void calculate_nonzeros_per_row_in_index_set(
+    std::shared_ptr<const DefaultExecutor> exec,
+    const matrix::Csr<ValueType, IndexType>* source,
+    const IndexSet<IndexType>& row_index_set,
+    const IndexSet<IndexType>& col_index_set, Array<IndexType>* row_nnz)
```

Comment:
Is there a reason for passing an array here?
```suggestion
    const IndexSet<IndexType>& col_index_set, IndexType* row_nnz)
```